### PR TITLE
decide track flow

### PR DESCRIPTION
**Pull request by jitwit**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -193,12 +197,12 @@
       "slug": "atbash-cipher",
       "uuid": "82169e85-3938-48da-ada9-7394adadc57a",
       "core": false,
-      "unlocked_by": "rna-transcription",
+      "unlocked_by": "hamming",
       "difficulty": 4,
```
Comment (jitwit) [inline config.json]: I made it so that several exercises are unlocked by earlier ones. I think that'll be more interesting for students and mentors alike

Comment (jitwit) [reply config.json]: This example I choose to comment on, of course, is an exception haha...

File: code/track.ss [hunk 2]
```diff
@@ -28,17 +28,17 @@
...
+		 exercisms)
+	   =>
+	   (lambda (config)
+	     (unless (assoc 'uuid config)
+	       (error 'check-config-for
+		      "please set uuid"
+		      problem))))
+	  (else (error 'check-config-for
+		       "please add problem to config/config.ss"
+		       problem)))))
```
Comment (jitwit) [inline code/track.ss]: All I did was visit the file... This is why I was against making so many whitespace fixes in the json pull

Comment (guygastineau) [reply code/track.ss]: That is terrifying.  Although, all I am seeing are the formatting diff sections, but that is just from this view.

You setup doesn't automatically add trailing whitespace does it?

Comment (guygastineau) [reply code/track.ss]: So, why does the formatting look like this now?

You are saying that the formatting in your editor looks like it does on the left, but it looks like the right hand side of the diff here on GitHub?

File: config.json [hunk 3]
```diff
@@ -193,12 +197,12 @@
...
       "topics": [
         "cipher",
         "string"
       ]
     }
   ]
-}
\ No newline at end of file
+}
```
Comment (guygastineau) [inline config.json]: Are you still using configlet to format the config?

Comment (guygastineau) [reply config.json]: Oh, does `make -j` run `configlet` ?

File: code/config.ss [hunk 4]
```diff
@@ -171,7 +174,7 @@
      ((slug . atbash-cipher)
```
Comment (guygastineau) [inline code/config.ss]: Why are there no formatting problems for us on the `cofig.ss` file?

File: config.json (hunk 3 above)
Comment (jitwit) [reply config.json]: I just branched from upstream/feature after getting yours in. Mm, answer is yes, I'll change the makefile in this pull

**Comment by guygastineau** [comment]
This looks good to me.

**Comment by rdipardo** [comment]
> @jitwit @guygastineau Do you remember why you chose to deprecate certain exercises in this file?

I found this thread by following a link in [the forum][0], where the deprecation of robot-name seems to have caused a bit of confusion.

From the current state of the source tree, it looks like robot-name was never fully implemented, and joined the deprecated list once it had been given up on. I've [made some headway in polishing it up][1], but haven't sought any direction yet, since I'm also waiting to learn if the deprecation was motivated by more than a simple loss of interest in maintaining it. My impression is that it's a rather trifling exercise, so official deprecation might just be the best thing for all concerned.

[0]: https://forum.exercism.org/t/select-featured-exercises-for-summer-of-sexps/5520/38
[1]: https://github.com/rdipardo/exercism-scheme/commits/wip/robot-name-redux

**Pull request merged**